src/Modal.js

```diff
@@ -315,4 +349,37 @@ const Modal = React.createClass({
   }
 });
 
-export default Modal;
+const ModalOverlay = React.createClass({
+  propTypes: {
+    ...Portal.propTypes
+  },
+
+  defaultProps: {
+    show: null
+  },
+
+  render() {
+    let { show, ...props } = this.props;
+
+    let modal = (
+      <Modal {...props}>{this.props.children}</Modal>
+    );
+    // I can't think of another way to not break back compat while defaulting container
+    if ( show != null ){
+      return (
+        <Portal show={show} container={props.container} >
+          { modal }
+        </Portal>
+      );
+    } else {
+      return modal;
+    }
+  }
+});
+
+ModalOverlay.Body = Body;
```

Comment:
These are most likely not going to work with factories when declared with this syntax. Do we need them? Not sure of the benefit of typing `Modal.Body` rather than `ModalBody`.